types/reactstrap/lib/Badge.d.ts

```diff
@@ -1,12 +1,12 @@
 import { CSSModule } from '../index';
 
-interface Props {
+export interface BadgeProps {
   color?: string;
   pill?: boolean;
   tag?: React.ReactType;
   className?: string;
   cssModule?: CSSModule;
 }
 
-declare var Badge: React.StatelessComponent<Props>;
+export const Badge: React.StatelessComponent<BadgeProps>;
```

Comment:
Unfortunately, this isn't correct either; there are no named exports in `Badge.d.ts` and the like; they're all default exports.

Reply:
I must have misunderstood what you were talking about then.

The default exports were left untouched in case someone else was referencing directly (it's still there, the diff isn't showing it). I added in named exports for the component in the file, as well as it's props.

What should it look like? Should the default export should be the only export in the file? If so, would the following be correct?
```
declare namespace Badge {
  interface Props {}
}
declare const Badge: React.StatelessComponent<Badge.Props>
export default Badge;
```

Reply:
Yup, sorry I didn't get back to this sooner. That is more appropriate, but you can't have `Props` as a separate type as well:

```ts
export interface Props {

}

const Badge: React.StatelessComponent<Props>;
export default Badge;
```

Reply:
I don't have an issue going through and updating everything, I just don't understand as to why it's incorrect / bad practice. Do you have an example of where the current definition could lead to issues?

What do you mean by: "you can't have `Props` as a separate type as well"? I don't understand the functional difference between your example and the current merge (unless you were saying what you couldn't do).

Reply:
If you had the following

```ts
export const Badge: React.StatelessComponent<Props>;
export default Badge;
```

That means that you also have a named export which doesn't actually exist, so you could write something like

```ts
import BadgeDefault, { BadgeNamed } from "reactstrap/lib/Badge";

let a = <BadgeDefault /> // okay, should work
let b = <BadgeNamed /> // error at runtime
```

Reply:
If the named export would cause run-time issues, then how come the tests are passing? When I re-exported everything in the index file, it's using the named export, not the default export.

I've also dropped the change into an existing project and it hasn't caused any issues. I can make a small test repo with these changes if that would help.

Reply:
The tests are passing for two reasons:

1. They don't test the runtime behavior.
2. There are no tests that import from anywhere other than the `index.d.ts` entry-point. For consumers who only consume from the entry-point, they'll never see a difference.

The `.d.ts` files themselves previously reflected [the way that the library was authored](https://github.com/reactstrap/reactstrap/blob/57d07e36b43b0b8197606fc884d3733fa25842d4/src/Breadcrumb.js#L36) - that each module within `lib` had a `default` export. For anyone who used to write

```ts
import Breadcrumb from "reactstrap/lib/Breadcrumb"
```

they'll simply get an error now. When they try to change it to

```ts
import { Breadcrumb } from "reactstrap/lib/Breadcrumb"
```

TypeScript won't report an error, but they'll get an error at runtime.

Reply:
Ohh, okay. That makes sense now; I wasn't sure what you were getting at before. I'll have a fix shortly. Thanks for the clarification, I really appreciate it!